compiler/rustc_middle/src/ty/abstract_const.rs

```diff
@@ -19,28 +19,32 @@ rustc_index::newtype_index! {
 ///
 /// This is only able to represent a subset of `MIR`,
 /// and should not leak any information about desugarings.
-#[derive(Debug, Clone, Copy)]
+#[derive(Hash, Debug, Clone, Copy, Ord, PartialOrd, Eq, PartialEq)]
 pub struct AbstractConst<'tcx> {
     // FIXME: Consider adding something like `IndexSlice`
     // and use this here.
     inner: &'tcx [Node<'tcx>],
-    substs: SubstsRef<'tcx>,
+    // FIXME: get rid of substs in this pr
```

Comment:
get rid of `AbstractConst` entirely, you can add `fn bound_(thir)_abstract_const` to the `TyCtxt` returing `Result<EarlyBinder<ty::Const<'tcx>>, ErrorGuaranteed>` an then getting the abstract const for an unevaluated one is `tcx.bound_abstract_const(ct.def.did)?.subst(tcx, substs)`.

We don't have to erase regions anymore I think

Reply:
is there any benefit to having the subst performed at the call site? I think the type will be quite long:
`Result<EarlyBinder<Option<ty::Const<'tcx>>, ErrorGuaranteed.

Reply:
`Result<Option<EarlyBinder<ty::Const<'tcx>>>, ErrorGuaranteed>` ^^

it's how similar queries are used, e.g. `bound_type_of` and `bound_fn_sig`. The reason for that is that fetching the data and substituting it feel distinct enough that keeping them split feels easier to understand to me.

Reply:
I tried removing the region erasing, but it still seems to be necessary.